### PR TITLE
More updates to support testing and regression on ATS-2.

### PR DESCRIPTION
**Pull request by KineticTheory**
### Background

* Tests were running about 100X too slow on rzansel because the test launcher wasn't configured correctly.

### Purpose of Pull Request

* [Fixes Redmine Issue #1945](https://rtt.lanl.gov/redmine/issues/1945)
* [Fixes Redmine Issue #1933](https://rtt.lanl.gov/redmine/issues/1933)

### Description of changes

+ Switch back to `lrun --pack` for launching jobs on ATS-2.
+ Set lrun as the default launcher for ATS-2.
+ In the Draco developer environment for ATS-2, make `cuda/11.0.2` the default.
+ Ensure that the c4 tests `tstOMP_API_off` and `tstOMP_API_on` are registered as having OpenMP enabled. In the `_on` test, change the test to use 39 OpenMP threads instead of 51.  ATS-2 throws an error if more than 40 threads are requested.

### Status

* Reference:
  * [Pre-Merge Code Review](https://github.com/lanl/Draco/wiki/Style-Guide)
  * [CDash Status](https://rtt.lanl.gov/cdash/index.php?project=Draco&filtercount=1&showfilters=1&field1=buildname&compare1=63&value1=-pr)
* Checks
  * [x] Travis/Appveyor CI checks pass
  * [x] Code coverage does not decrease
  * [x] [Valgrind test passes](https://rtt.lanl.gov/CDash/index.php?project=Draco)
  * [x] [Toss3 checks pass](https://rtt.lanl.gov/CDash/index.php?project=Draco)
  * [x] [Trinitite checks pass](https://rtt.lanl.gov/CDash/index.php?project=Draco)
  * [x] Code reviewed/approved, sufficient DbC checks, testing, documentation

### CODE REVIEW

**Comment by KineticTheory** [comment]
@alexrlongne Do you have time to review & merge this PR.  If fixes several issues wrt running tests on ATS-2.

File: config/setupMPI.cmake [hunk 1]
```diff
@@ -399,7 +399,12 @@ macro( setupSpectrumMPI )
       # 1 resource set; OMP_NUM_THREADS tasks; no gpu; no binding
       set( MPIEXEC_OMP_PREFLAGS "--nrs 1 -c $ENV{OMP_NUM_THREADS} -g 0 --bind none")
     elseif( MPIEXEC_EXECUTABLE MATCHES lrun )
-      set( MPIEXEC_OMP_PREFLAGS "-c $ENV{OMP_NUM_THREADS} --threads=$ENV{OMP_NUM_THREADS} --bind=off -v" )
+      if( DEFINED ENV{OMP_NUM_THREADS} )
+        # --bind=off
+        set( MPIEXEC_OMP_PREFLAGS "--pack --threads=$ENV{OMP_NUM_THREADS} -c$ENV{OMP_NUM_THREADS} -v" )
```
Comment (alexrlongne) [inline config/setupMPI.cmake:404]: It seems odd to have the same number of cores and threads, how does lrun interpret this?

File: config/setupMPI.cmake [hunk 2]
```diff
@@ -399,7 +399,12 @@ macro( setupSpectrumMPI )
...
+      else()
+        set( MPIEXEC_OMP_PREFLAGS "--pack --threads=4 -c 4 -v" )
```
Comment (alexrlongne) [inline config/setupMPI.cmake:406]: Is this `--threads=4` used by pthreads in the worker model?

File: config/setupMPI.cmake (hunk 1 above)
Comment (KineticTheory) [reply config/setupMPI.cmake:404]: The `-v` option will print the `jsrun` command.  I tested these types of commands and they looked correct.  
```
lrun -n 2 --pack --threads=1 -v js_task_info # no OpenMP
lrun -n 2 --pack --threads=10 -c 10 -v js_task_info  # with 10 OpenMP threads
```
Can you repeat this test and advise me?  This appeared to do what I wanted, but maybe I misunderstood the output.

File: config/setupMPI.cmake (hunk 2 above)
Comment (KineticTheory) [reply config/setupMPI.cmake:406]: If you don't specify this option, lrun generates warnings about using more threads that requested/allowed.  I think this is the default of 4 hw threads per core.  I would like your input as all of these options are a bit arcane to me.  The hard part is creating commands that allow us to pack a node.  Prior to these changes smpi was running rank 0 of all mpi jobs on core 0.  That was very bad.

File: config/setupMPI.cmake (hunk 1 above)
Comment (alexrlongne) [reply config/setupMPI.cmake:404]: Looking at the jsrun visualizer I think these commands are doing the right thing. The "threads" command seems to just be for OpenMP and not "threads per core".

File: config/setupMPI.cmake (hunk 2 above)
Comment (alexrlongne) [reply config/setupMPI.cmake:406]: OK, I think I get what's happening, thanks.

**Pull request merged**